CMakeLists.txt

```diff
@@ -0,0 +1,200 @@
+SET(PACKAGE osm2pgsql)
+SET(PACKAGE_NAME osm2pgsql)
+SET(PACKAGE_VERSION 0.89.0-dev)
+
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.7)
+
+PROJECT(osm2pgsql)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
+SET(DATA_DIR \".\")
+
+OPTION(TEST_BUILD "Build with tests and strict options" OFF)
+
+if (NOT WIN32 AND NOT APPLE)
+  # No need for this path, just a workaround to make cmake check pass on all systems
+  set(PostgreSQL_TYPE_INCLUDE_DIR /usr/include)
+endif()
+
+# Just in case user installed RPMs from http://yum.postgresql.org/
+list(APPEND PostgreSQL_ADDITIONAL_SEARCH_PATHS /usr/pgsql-9.3 /usr/pgsql-9.4)
+
+if ("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
+  message(FATAL_ERROR "In-source builds are not allowed, please use a separate build directory like `mkdir build && cd build && cmake ..`")
+endif()
+
+message(STATUS "Building osm2pgsql ${PACKAGE_VERSION}")
+
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE=Release)
+endif()
+
+if ( MSVC )
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS -DNOMINMAX -wd4996)
+  set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:30000000")
+else()
+  add_definitions(-Wall)
+  add_definitions(-std=c++11)
+  if (TEST_BUILD)
+    if (NOT APPLE)
+      add_definitions(-Werror)
+    endif()
+    add_definitions(-pedantic)
+  endif()
+endif()
+
+option(EXTERNAL_LIBOSMIUM "Do not use the bundled libosmium" OFF)
+
+#############################################################
+# Detect available headers and set global compiler options
+#############################################################
+
+INCLUDE (CheckIncludeFiles)
+INCLUDE (CheckFunctionExists)
+INCLUDE (CheckTypeSize)
+
+ADD_DEFINITIONS( -DOSM2PGSQL_DATADIR=${DATA_DIR} )
+ADD_DEFINITIONS( -DFIXED_POINT )
+
+CHECK_INCLUDE_FILES (termios.h HAVE_TERMIOS_H)
+CHECK_INCLUDE_FILES (libgen.h HAVE_LIBGEN_H)
+CHECK_INCLUDE_FILES (unistd.h HAVE_UNISTD_H)
+CHECK_INCLUDE_FILES (sys/wait.h HAVE_SYS_WAIT_H)
+CHECK_INCLUDE_FILES (sys/time.h HAVE_SYS_TIME_H)
+CHECK_INCLUDE_FILES (sys/mman.h HAVE_MMAP)
+
+if (WIN32)
+  set(HAVE_LIBGEN_H FALSE)
+endif()
+
+CHECK_FUNCTION_EXISTS(lseek64 HAVE_LSEEK64)
+CHECK_FUNCTION_EXISTS(posix_fallocate HAVE_POSIX_FALLOCATE)
+CHECK_FUNCTION_EXISTS(posix_fadvise HAVE_POSIX_FADVISE)
+CHECK_FUNCTION_EXISTS(sync_file_range HAVE_SYNC_FILE_RANGE)
+CHECK_FUNCTION_EXISTS(fork HAVE_FORK)
+
+CHECK_TYPE_SIZE("off_t" SIZEOF_OFF_T)
+
+if (NOT WIN32 AND NOT APPLE AND NOT HAVE_LSEEK64 AND NOT SIZEOF_OFF_T EQUAL 8)
+  message(FATAL_ERROR "Flat nodes cache requires a 64 bit capable seek")
+endif()
+
+#############################################################
+# Find necessary libraries
+#############################################################
+
+if (NOT EXTERNAL_LIBOSMIUM)
+  set(OSMIUM_INCLUDE_DIR "contrib/libosmium")
```

Comment:
This should become "${CMAKE_CURRENT_SOURCE_DIR}/contrib/libosmium" so it is an absolute path.